Doesn't work on firefox. Chrome works fine

It seems that selkies-gstreamer will stuck on `Waiting for video stream ` on Firefox 96.

The status message only shows 
``
[19:0:0] [signalling] Connecting to server.
[19:0:2] [signalling] Registering with server, peer ID: 1
[19:0:4] [signalling] Registered with server.
[19:0:4] [signalling] Waiting for video stream.
[19:0:6] [webrtc] [ERROR] attempt to send data channel message before channel was open.
[19:0:6] [webrtc] Received incoming video stream from peer
[19:0:6] [webrtc] Received incoming audio stream from peer
[19:0:7] [webrtc] [ERROR] attempt to send data channel message before channel was open.

``

--- a/Dockerfile.example
+++ b/Dockerfile.example
@@ -105,7 +105,7 @@ RUN pip3 install /opt/${PYPI_PACKAGE}-${PACKAGE_VERSION}-py3-none-any.whl
 # Remove this and uncommend line in setup.cfg after this PR is merged and in release:
 # https://github.com/python-xlib/python-xlib/pull/218
 RUN if [ "${UBUNTU_RELEASE}" != "18.04" ] ; then \
-    pip3 install -e git+https://github.com/selkies-project/python-xlib.git@add-xfixes-cursor#egg=python-xlib ; fi
+    pip3 install --upgrade --force-reinstall https://github.com/python-xlib/python-xlib/archive/master.zip ; fi
 
 # Setup global bashrc to configure gstreamer environment
 RUN echo 'export DISPLAY=:0' \

--- a/addons/gstreamer/Dockerfile
+++ b/addons/gstreamer/Dockerfile
@@ -32,7 +32,8 @@ RUN \
         libwebrtc-audio-processing-dev \
         libssl-dev \
         libsrtp2-dev \
-        libx264-dev && \
+        libx264-dev \
+        libvpx-dev && \
     rm -rf /var/lib/apt/lists/*
 
 # Install meson and ninja

--- a/src/selkies_gstreamer/__main__.py
+++ b/src/selkies_gstreamer/__main__.py
@@ -497,8 +497,8 @@ def main():
     logger.info("initial server RTC config: {}".format(rtc_config))
 
     # Extract args
-    enable_audio = args.enable_audio == "true"
-    enable_resize = args.enable_resize == "true"
+    enable_audio = args.enable_audio.lower() == "true"
+    enable_resize = args.enable_resize.lower() == "true"
     curr_fps = int(args.framerate)
     curr_video_bitrate = int(args.video_bitrate)
     curr_audio_bitrate = int(args.audio_bitrate)

--- a/src/selkies_gstreamer/gstwebrtc_app.py
+++ b/src/selkies_gstreamer/gstwebrtc_app.py
@@ -901,11 +901,17 @@ class GSTWebRTCApp:
         self.webrtcbin.emit('set-local-description', offer, promise)
         promise.interrupt()
         loop = asyncio.new_event_loop()
+        # Firefox needs profile-level-id=42e01f in the offer, but webrtcbin does not add this.
+        # Remove when fixed in webrtcbin.
+        #   https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/1106
         sdp_text = offer.sdp.as_text()
-        if '264' in self.encoder and 'profile-level-id' not in sdp_text:
-            # Workaround to force high h264 profile, fix for Firefox.
-            logger.warning("injecting profile-level-id to SDP")
-            sdp_text = sdp_text.replace('packetization-mode=1', 'profile-level-id=42e01f;packetization-mode=1')
+        if '264' in self.encoder:
+            if 'profile-level-id' not in sdp_text:
+                logger.warning("injecting profile-level-id to SDP")
+                sdp_text = sdp_text.replace('packetization-mode=1', 'profile-level-id=42e01f;packetization-mode=1')
+            if 'level-asymmetry-allowed' not in sdp_text:
+                logger.warning("injecting level-asymmetry-allowed to SDP")
+                sdp_text = sdp_text.replace('packetization-mode=1', 'level-asymmetry-allowed=1;packetization-mode=1')
         loop.run_until_complete(self.on_sdp('offer', sdp_text))
 
     def __on_negotiation_needed(self, webrtcbin):

--- a/src/selkies_gstreamer/gstwebrtc_app.py
+++ b/src/selkies_gstreamer/gstwebrtc_app.py
@@ -901,7 +901,12 @@ class GSTWebRTCApp:
         self.webrtcbin.emit('set-local-description', offer, promise)
         promise.interrupt()
         loop = asyncio.new_event_loop()
-        loop.run_until_complete(self.on_sdp('offer', offer.sdp.as_text()))
+        sdp_text = offer.sdp.as_text()
+        if '264' in self.encoder and 'profile-level-id' not in sdp_text:
+            # Workaround to force high h264 profile, fix for Firefox.
+            logger.warning("injecting profile-level-id to SDP")
+            sdp_text = sdp_text.replace('packetization-mode=1', 'profile-level-id=42e01f;packetization-mode=1')
+        loop.run_until_complete(self.on_sdp('offer', sdp_text))
 
     def __on_negotiation_needed(self, webrtcbin):
         """Handles on-negotiation-needed signal, generates create-offer action

--- a/src/selkies_gstreamer/webrtc_signalling.py
+++ b/src/selkies_gstreamer/webrtc_signalling.py
@@ -25,8 +25,8 @@ Interfaces with signalling server found at:
   https://github.com/centricular/gstwebrtc-demos/tree/master/signalling
 
     Usage example:
-    from webrtc_signalling import WebRTCSinalling
-    signalling = WebRTCSinalling(server, id, peer_id)
+    from webrtc_signalling import WebRTCSignalling
+    signalling = WebRTCSignalling(server, id, peer_id)
     signalling.on_connect = lambda: signalling.setup_call()
     signalling.connect()
     signalling.start()

